src/ServiceManager.php

```diff
@@ -737,6 +737,32 @@ private function createLazyServiceDelegatorFactory()
         return $this->lazyServicesDelegator;
     }
 
+    /**
+     * Merge delegators avoiding multiple same delegators for the same service.
+     * It works with strings and class instances.
+     * It's not possible to de-duple anonymous functions
+     *
+     * @param string[][]|Factory\DelegatorFactoryInterface[][] $config
+     * @return string[][]|Factory\DelegatorFactoryInterface[][]
+     */
+    private function mergeDelegators(array $config)
+    {
+        foreach ($config as $key => $delegators) {
+            if (! isset($this->delegators[$key])) {
```

Comment:
This should probably be an `array_key_exists()` check to prevent `isset()` semantics (`isset()` can work with undefined variables, which may happen when refactoring at a later point in time)